apps/api/src/app/widgets/e2e/get-subscriber-preference.e2e.ts

```diff
@@ -61,6 +67,18 @@ describe('GET /widget/preferences', function () {
 
     expect(data.preference.channels.email).to.equal(false);
     expect(data.preference.channels.in_app).to.equal(false);
+
+    expect(data.preference.overrides.find((sources) => sources.channel === 'email').source).to.equal('subscriber');
+    expect(data.preference.overrides.find((sources) => sources.channel === 'in_app').source).to.equal('template');
+  });
+
+  it('should filter not active channels and sources', async function () {
+    const response = await getSubscriberPreference(session.subscriberToken);
+
+    const data = response.data.data[0];
+
+    expect(Object.keys(data.preference.channels).length).to.equal(2);
+    expect(data.preference.overrides.length).to.equal(2);
```

Comment:
can't we combine this one with `should fetch a default user preference` ?

Reply:
We could but I wanted a separate test that points to the filtering of the active channels.